### ISSUE TITLE
Can't assign group to a child entity directly from LDAP import

### ISSUE THREAD

**Issue by 0xNath** [glpi-project/glpi#12056]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.2

### Bug description

I have 4 levels of entities.
let say that I set my user to the top entitie and I want to import a group to an under entitie :
<img width="832" alt="image" src="https://user-images.githubusercontent.com/67143274/176642344-f4ca3ded-6f1b-4ddd-bbf0-ecac5edccb48.png">

When I change the entities of destination of the groups and do the import, they go to the entity that my user has currently :
![image](https://user-images.githubusercontent.com/67143274/176649413-602ffc46-914a-4b6c-a980-888239e6e809.png)

### Page URL

https://localhost/front/ldap.group.import.php?next=servers?mode=0&action=show

### Steps To reproduce

1. setup GLPI
2. add an LDAP authentification method
3. go the the group pages
4. click on the "LDAP directory link button"
5. click on button "Import of new groups"
6. Select the previously authentification method in the list
7. Select two random groups, change the "destination entity" of one of the groups to a different entity that your user is set to
8. When going into the groups page, you should see that the two groups are under the same entity

### Your GLPI setup information

<details><summary>Informations sur le système, l'installation et la configuration</summary><pre>GLPI 10.0.2 ( => C:\xampp\htdocs\glpi)
Installation mode: TARBALL
Current language:fr_FR

</details><details><summary>Server</summary><pre> 
Operating system: Windows NT serverName 10.0 build 17763 (Windows Server 2019) AMD64
PHP 8.1.6 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, bcmath, bz2, calendar,
	ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, intl, json, ldap, libxml, mbstring, mysqli,
	mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, readline, session, standard, tokenizer, xml, xmlreader, xmlwriter, zip, zlib)
Setup: max_execution_time="600" memory_limit="512M" post_max_size="40M" safe_mode="" session.save_handler="files"
	upload_max_filesize="40M" 
Software: Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6 (Apache/2.4.53 (Win64) OpenSSL/1.1.1n PHP/8.1.6 Server at
	serverName.domain.local Port 443
)
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36 Edg/103.0.1264.37
Server Software: mariadb.org binary distribution
	Server Version: 10.4.24-MariaDB
	Server SQL Mode: NO_ZERO_IN_DATE,NO_ZERO_DATE,NO_ENGINE_SUBSTITUTION
	Parameters: glpi@localhost/glpi
	Host info: localhost via TCP/IP
	
PHP version (8.1.6) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, json, simplexml.
curl extension is installed.
gd extension is installed.
intl extension is installed.
libxml extension is installed.
zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (10.4.24) is supported.
The log file has been created successfully.
Write access to C:\glpi\files/_cache has been validated.
Write access to C:\glpi\config has been validated.
Write access to C:\glpi\files/_cron has been validated.
Write access to C:\glpi\files has been validated.
Write access to C:\glpi\files/_dumps has been validated.
Write access to C:\glpi\files/_graphs has been validated.
Write access to C:\glpi\files/_lock has been validated.
Write access to C:\glpi\files/_pictures has been validated.
Write access to C:\glpi\files/_plugins has been validated.
Write access to C:\glpi\files/_rss has been validated.
Write access to C:\glpi\files/_sessions has been validated.
Write access to C:\glpi\files/_tmp has been validated.
Write access to C:\glpi\files/_uploads has been validated.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
zip extension is installed.
bz2 extension is installed.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring.
Following extensions are not present: sodium.
Write access to C:\xampp\htdocs\glpi/marketplace has been validated.
Timezones seems loaded in database.

</details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: C:\xampp\htdocs\glpi
GLPI_CONFIG_DIR: C:\glpi\config
GLPI_VAR_DIR: C:\glpi\files
GLPI_LOG_DIR: C:\glpi\logs
GLPI_MARKETPLACE_DIR: C:\xampp\htdocs\glpi/marketplace
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_INSTALL_MODE: TARBALL
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: 1
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_CENTRAL_WARNINGS: 1
GLPI_DOC_DIR: C:\glpi\files
GLPI_CACHE_DIR: C:\glpi\files/_cache
GLPI_CRON_DIR: C:\glpi\files/_cron
GLPI_DUMP_DIR: C:\glpi\files/_dumps
GLPI_GRAPH_DIR: C:\glpi\files/_graphs
GLPI_LOCAL_I18N_DIR: C:\glpi\files/_locales
GLPI_LOCK_DIR: C:\glpi\files/_lock
GLPI_PICTURE_DIR: C:\glpi\files/_pictures
GLPI_PLUGIN_DOC_DIR: C:\glpi\files/_plugins
GLPI_RSS_DIR: C:\glpi\files/_rss
GLPI_SESSION_DIR: C:\glpi\files/_sessions
GLPI_TMP_DIR: C:\glpi\files/_tmp
GLPI_UPLOAD_DIR: C:\glpi\files/_uploads
GLPI_INVENTORY_DIR: C:\glpi\files/_inventories
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: C:\xampp\htdocs\glpi/locales
GLPI_VERSION: 10.0.2
GLPI_SCHEMA_VERSION: 10.0.2@a130db99c7d9b131c2e2ea59fe0d6260fe93d831
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_MIN_PHP: 7.4.0
GLPI_MAX_PHP: 8.2.0
GLPI_YEAR: 2022

</details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.6 in (C:\xampp\htdocs\glpi\vendor\htmlawed\htmlawed)
phpmailer/phpmailer version 6.6.0 in (C:\xampp\htdocs\glpi\vendor\phpmailer\phpmailer\src)
simplepie/simplepie version 1.5.8 in (C:\xampp\htdocs\glpi\vendor\simplepie\simplepie\library)
tecnickcom/tcpdf version 6.4.4 in (C:\xampp\htdocs\glpi\vendor\tecnickcom\tcpdf)
michelf/php-markdown in (C:\xampp\htdocs\glpi\vendor\michelf\php-markdown\Michelf)
true/punycode in (C:\xampp\htdocs\glpi\vendor\true\punycode\src)
iamcal/lib_autolink in (C:\xampp\htdocs\glpi\vendor\iamcal\lib_autolink)
sabre/dav in (C:\xampp\htdocs\glpi\vendor\sabre\dav\lib\DAV)
sabre/http in (C:\xampp\htdocs\glpi\vendor\sabre\http\lib)
sabre/uri in (C:\xampp\htdocs\glpi\vendor\sabre\uri\lib)
sabre/vobject in (C:\xampp\htdocs\glpi\vendor\sabre\vobject\lib)
laminas/laminas-i18n in (C:\xampp\htdocs\glpi\vendor\laminas\laminas-i18n\src)
laminas/laminas-servicemanager in (C:\xampp\htdocs\glpi\vendor\laminas\laminas-servicemanager\src)
monolog/monolog in (C:\xampp\htdocs\glpi\vendor\monolog\monolog\src\Monolog)
sebastian/diff in (C:\xampp\htdocs\glpi\vendor\sebastian\diff\src)
donatj/phpuseragentparser in (C:\xampp\htdocs\glpi\vendor\donatj\phpuseragentparser\src\UserAgent)
elvanto/litemoji in (C:\xampp\htdocs\glpi\vendor\elvanto\litemoji\src)
symfony/console in (C:\xampp\htdocs\glpi\vendor\symfony\console)
scssphp/scssphp in (C:\xampp\htdocs\glpi\vendor\scssphp\scssphp\src)
laminas/laminas-mail in (C:\xampp\htdocs\glpi\vendor\laminas\laminas-mail\src\Protocol)
laminas/laminas-mime in (C:\xampp\htdocs\glpi\vendor\laminas\laminas-mime\src)
rlanvin/php-rrule in (C:\xampp\htdocs\glpi\vendor\rlanvin\php-rrule\src)
blueimp/jquery-file-upload in (C:\xampp\htdocs\glpi\vendor\blueimp\jquery-file-upload\server\php)
ramsey/uuid in (C:\xampp\htdocs\glpi\vendor\ramsey\uuid\src)
psr/log in (C:\xampp\htdocs\glpi\vendor\psr\log\Psr\Log)
psr/simple-cache in (C:\xampp\htdocs\glpi\vendor\psr\simple-cache\src)
psr/cache in (C:\xampp\htdocs\glpi\vendor\psr\cache\src)
league/csv in (C:\xampp\htdocs\glpi\vendor\league\csv\src)
mexitek/phpcolors in (C:\xampp\htdocs\glpi\vendor\mexitek\phpcolors\src\Mexitek\PHPColors)
guzzlehttp/guzzle in (C:\xampp\htdocs\glpi\vendor\guzzlehttp\guzzle\src)
guzzlehttp/psr7 in (C:\xampp\htdocs\glpi\vendor\guzzlehttp\psr7\src)
glpi-project/inventory_format in (C:\xampp\htdocs\glpi\vendor\glpi-project\inventory_format\lib\php)
wapmorgan/unified-archive in (C:\xampp\htdocs\glpi\vendor\wapmorgan\unified-archive\src)
paragonie/sodium_compat in (C:\xampp\htdocs\glpi\vendor\paragonie\sodium_compat\src)
symfony/cache in (C:\xampp\htdocs\glpi\vendor\symfony\cache)
html2text/html2text in (C:\xampp\htdocs\glpi\vendor\html2text\html2text\src)
symfony/css-selector in (C:\xampp\htdocs\glpi\vendor\symfony\css-selector)
symfony/dom-crawler in (C:\xampp\htdocs\glpi\vendor\symfony\dom-crawler)
twig/twig in (C:\xampp\htdocs\glpi\vendor\twig\twig\src)
twig/string-extra in (C:\xampp\htdocs\glpi\vendor\twig\string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 not found
symfony/polyfill-php81 not found

</details><details><summary>LDAP directories</summary><pre> 
Server: '10.90.16.25', Port: '389', BaseDN: 'OU=XXX1,OU=BWT,DC=domain,DC=local', Connection filter:
		'(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2)))', RootDN:
		'glpi@domain.com', Use TLS: none
Server: '10.90.32.25', Port: '389', BaseDN: 'OU=XXX1,OU=BWT,DC=domain,DC=local', Connection filter:
		'(&(objectClass=user)(objectCategory=person)(!(userAccountControl:1.2.840.113556.1.4.803:=2)))', RootDN:
		'glpi@domain.com', Use TLS: none

</details><details><summary>SQL replicas</summary><pre> 
Not active

</details><details><summary>Notifications</summary><pre> 
Way of sending emails: PHP

</details><details><summary>Plugins list</summary><pre> 
	activity             Name: Activités quotidiennes         Version: 3.1.1      State: Installed / not activated               
		Install Method: Marketplace
	addressing           Name: Adressage IP                   Version: 3.0.0      State: Installed / not activated               
		Install Method: Marketplace
	additionalalerts     Name: Alertes supplémentaires        Version: 2.4.0      State: Installed / not activated               
		Install Method: Marketplace
	webapplications      Name: Applications Web               Version: 4.0.1      State: Installed / not activated               
		Install Method: Marketplace
	barcode              Name: Barcode                        Version: 2.7.0      State: Installed / not activated               
		Install Method: Marketplace
	camerainput          Name: Camera Input                   Version: 2.0.0      State: Installed / not activated               
		Install Method: Marketplace
	positions            Name: Cartographie                   Version: 6.0.1      State: Installed / not activated               
		Install Method: Marketplace
	fields               Name: Champs supplémentaires         Version: 1.16.0     State: Installed / not activated               
		Install Method: Marketplace
	accounts             Name: Comptes                        Version: 3.0.2      State: Installed / not activated               
		Install Method: Marketplace
	printercounters      Name: Compteurs Imprimantes          Version: 2.0.0      State: Installed / not activated               
		Install Method: Marketplace
	datainjection        Name: Data Injection                 Version: 2.11.2     State: Error / to clean                        
		Install Method: Manual
	gantt                Name: gantt                          Version: 1.0.0      State: Installed / not activated               
		Install Method: Marketplace
	genericobject        Name: Gestion d'objets               Version: 2.12.1     State: Installed / not activated               
		Install Method: Marketplace
	order                Name: Gestion des commandes          Version: 2.9.0      State: Installed / not activated               
		Install Method: Marketplace
	glpiinventory        Name: GLPI Inventory                 Version: 1.0.3      State: Enabled                                 
		Install Method: Marketplace
	manufacturersimports Name: Imports fabricants             Version: 3.0.1      State: Installed / not activated               
		Install Method: Marketplace
	mreporting           Name: Plus de rapports               Version: 1.8.0      State: Installed / not activated               
		Install Method: Marketplace
	resources            Name: Ressources humaines            Version: 3.0.2      State: Installed / not activated               
		Install Method: Marketplace

</pre></details>

### Anything else?

A temporary solution is to import the groups by changing the entity of the user

**Comment by 0xNath** [comment]
This is the code fom `AuthLDAP.php` approximately line 315:

```
if (isset($input["ldap_import_entities"][$id])) {
    $entity = $input["ldap_import_entities"][$id];
} else {
     $entity = $_SESSION["glpiactive_entity"];
}
```

In my case we seems to always fall back into the `else`  statement and use the session entity ID.

**Comment by cconard96** [comment]
Hello,
Could you test the fix proposed in #12068?

**Comment by 0xNath** [comment]
It's working like a charm ! Thank you ! 👍🏻
